src/MICore/DebuggerDisposedException.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace MICore
+{
+    /// <summary>
+    /// Exception thrown to specify the debugger is disposed.
+    /// </summary>
+    public class DebuggerDisposedException : ObjectDisposedException
+    {
+        /// <summary>
+        /// Command where the abort happened.
+        /// </summary>
+        public string AbortedCommand { get; private set; }
+
+        private const string Debugger = "Debugger";
+
+        /// <summary>
+        /// Constructor for the DebuggerDisposedException which takes the aborted command.
+        /// </summary>
+        /// <param name="abortedCommand">MI Command</param>
+        public DebuggerDisposedException(string abortedCommand = null) : base(Debugger)
+        {
+            AbortedCommand = abortedCommand;
+        }
+
+        /// <summary>
+        /// Constructor for the DebuggerDisposedException which takes message, innerException and aborted command.
+        /// </summary>
+        public DebuggerDisposedException(string message, Exception innerException, string abortedCommand = null) : base(message, innerException)
+        {
+            AbortedCommand = abortedCommand;
+        }
+
+        /// <summary>
+        /// Constructor for the DebuggerDisposedException which takes message and aborted command.
+        /// </summary>
+        public DebuggerDisposedException(string message, string abortedCommand = null) : base(Debugger, message)
```

Comment:
Good question! I guess [https://msdn.microsoft.com/en-us/library/dd264739.aspx](https://msdn.microsoft.com/en-us/library/dd264739.aspx) explains at "Overload Resolution" 

---
In reply to: [75942444](https://github.com/Microsoft/MIEngine/pull/422#discussion_r75942444) [](ancestors = 75942444)